airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -455,17 +462,91 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
               AutoCloseableIterators.lazyIterator(incrementalIteratorSupplier, null)));
 
     } else if (PostgresUtils.isXmin(sourceConfig) && isIncrementalSyncMode(catalog)) {
-      final XminStateManager xminStateManager = new XminStateManager(stateManager.getRawStateMessages());
-      final PostgresXminHandler handler = new PostgresXminHandler(database, sourceOperations, getQuoteString(), xminStatus, xminStateManager);
-      return handler.getIncrementalIterators(catalog, tableNameToTable, emittedAt);
+      final List<AirbyteStateMessage> rawStateMessages = stateManager.getRawStateMessages();
+
+      final List<AirbyteStateMessage> statesFromCtidSync = new ArrayList<>();
+      final List<AirbyteStateMessage> statesFromXminSync = new ArrayList<>();
+
+      final Set<AirbyteStreamNameNamespacePair> alreadySeenStreams = new HashSet<>();
```

Comment:
Can we add helper methods to calculate `streamsStillInCtidSync`, `streamsForXminSync`, etc.